back/libs/account-fca/src/services/account-fca.service.ts

```diff
@@ -54,8 +56,8 @@ export class AccountFcaService {
       throw new CoreFcaAgentAccountBlockedException();
     }
 
-    if (!some(account.idpIdentityKeys, idpAgentKeys)) {
-      account.idpIdentityKeys.push(idpAgentKeys);
+    if (!some(account.idpIdentityKeys, idpFullKeys)) {
```

Comment:
issue: the algorithm should be:

if an idpIdentityKey exist with these specific idpUid and idpSub
then we update the email of the corresponding idpIdentityKeys
else we push the entire idpFullKeys in the list.

The problem here is that idpMail is used in the match.